tests_extractor.d

```diff
@@ -127,6 +127,24 @@ private:
         */
         outFile.writefln("# line %d", u.line > 2 ? u.line - 2 : 0);
 
+        static immutable predefinedAttributes = ["nogc", "system", "nothrow", "safe", "trusted", "pure"];
+
+        // write system attributes
+        foreach (attr; attributes)
+        {
+            const atAttribute = attr.atAttribute;
+            if (atAttribute is null)
+                continue;
+
+            const atText = atAttribute.identifier.text;
+
+            // ignore custom attributes (@myArg)
+            if (!predefinedAttributes.canFind(atText))
+                continue;
+
+            outFile.write(text("@", atText, " "));
```

Comment:
I don't think this would work with attributes whose syntax doesn't start with `@` , like `pure` and `nothrow`

Reply:
Yup, but DScanner treats these guys differently:

```
$ dscanner --ast foo.d | xmllint --format -
<?xml version="1.0"?>
<module>
  <declaration>
    <attribute attribute="pure"/>
    <attribute>
      <atAttribute>
        <identifier>safe</identifier>
      </atAttribute>
    </attribute>
    <unittest>
      <blockStatement>
</blockStatement>
    </unittest>
  </declaration>
</module>
```

Reply:
But yeah it's probably a good idea to extract these guys too while we're at it: https://github.com/dlang/tools/pull/371